src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,79 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE];
+        values = (V[]) new Object[MAX_SIZE];
+        size = 0;
+    }
+
+    public void putKeyIsNull(V value) {
+        for (int i = 0; i < size; i++) {
+            if (keys[i] == null) {
+                values[i] = value;
+                return;
+            }
+        }
```

Comment:
Let's extract it into a separate method, e.g like this:
```suggestion
         if (containsKey(key)) {
                values[indexOf(key)] = value;
                return;
        }
```

Don't duplicate the code. Try to create a method that checks that key is already present. Also, try to create a method that returns the index of the key, which will be used to store the value.

It will significantly simplify your code and reduce the number of repetitive lines.